### PR TITLE
Ch3d2 auto

### PR DESCRIPTION
**Pull request by robinnedergard**

### CODE REVIEW

File: Verification/Challenge/get.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+		request = {
+			url: request.content.details.url,
+			transactionId: request.content.details.data?.threeDSServerTransID ?? "",
+			acsTransactionID: request.content.details.data?.acsTransID ?? "",
+		}
+	const challengeNotificationUrl =
+		merchant.card.url.endsWith("7082") || merchant.card.url.endsWith("cardfunc.com")
+			? merchant.card.url + "/card/" + token + "/verification?mode=show&merchant=" + (merchant.card.id ?? merchant.sub)
+			: merchant.card.url + "/card/" + token + "/verification?mode=show"
+	let challengeData = authly.Base64.encode(
```
Comment (florianebmeier) [inline Verification/Challenge/get.ts:25]: Don't reuse variables

File: Verification/Challenge/get.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	merchant: model.Merchant & authly.Payload,
+	token: authly.Token
+): Promise<string | undefined> {
+	if (Challenge.is(request))
+		request = {
+			url: request.content.details.url,
+			transactionId: request.content.details.data?.threeDSServerTransID ?? "",
+			acsTransactionID: request.content.details.data?.acsTransID ?? "",
+		}
+	const challengeNotificationUrl =
```
Comment (florianebmeier) [inline Verification/Challenge/get.ts:21]: Keep variable close to where it is used. Only use is in line 50

File: Verification/Challenge/get.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+export namespace get {
+	export const command: paramly.Command<Connection> = {
+		name: "get",
+		description: "Performs challenge 3D. Only works with 3D simulator.",
+		examples: [
+			["<url> <transactionId> <acsTransID> <cardToken>", "Perform challenge 3D for given URL and transactionId."],
+		],
+		execute: async (connection, argument, flags) => {
+			const merchant = (await authly.Verifier.create("public").verify(
```
Comment (florianebmeier) [inline Verification/Challenge/get.ts:67]: Bump authly.

File: Verification/Method/get.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+	if (Method.is(request))
+		request = {
+			url: request.content.details.url,
+			transactionId: request.content.details.data?.threeDSServerTransID ?? "",
+		}
+	const methodNotificationUrl =
+		merchant.card.url.endsWith("7082") || merchant.card.url.endsWith("cardfunc.com")
+			? merchant.card.url + "/card/" + token + "/verification?mode=show&merchant=" + (merchant.card.id ?? merchant.sub)
+			: merchant.card.url + "/card/" + token + "/verification?mode=show"
+	let methodData = authly.Base64.encode(
```
Comment (florianebmeier) [inline Verification/Method/get.ts:24]: Dont reuse variables

**Review by florianebmeier — CHANGES_REQUESTED** [review]

**Review by florianebmeier — APPROVED** [review]

**Review by fredrikbryntesson — APPROVED** [review]

**Pull request merged**